synapse/tests/test_daemon.py

```diff
@@ -95,3 +96,19 @@ def __init__(self):
         self.assertEqual( dcon.addons.get('haha').foo(10,y=30), 40 )
 
         self.assertEqual( dcon.locs.get('blah').woot.foo(10,y=30), 40 )
+
+    def test_daemon_conf_onfini(self):
+
+        conf = {
+            'ctors': (
+                ('fini', 'ctor://synapse.eventbus.EventBus()'),
+            ),
+            'dmon:share': (
+                ('fini', {'onfini': True}),
+            ),
+        }
+
+        dmon = s_daemon.Daemon()
+        flen = len(dmon._fini_funcs)
+        dmon.loadDmonConf(conf)
+        self.assertEqual(len(dmon._fini_funcs), flen + 1)
```

Comment:
Lets modify this test to call fini() on dmon and then check assertTrue( dmon.shared.get('fini').isfini )

Also, in general for the unit tests, any EventBus instances *must* be fini'd within to test to prevent resource leaks.